connection.js

```diff
@@ -20,7 +21,7 @@ function Connection(duplex, opts) {
 
   opts = opts || {}
 
-  var inStream  = generateStream()
+  var inStream  = writeToStream()
```

Comment:
You don't need to create a stream. You can pass the original `duplex` directly to `writeToStream`.